docs/DeveloperGuide.md

```diff
@@ -280,17 +377,169 @@ Use case ends.
+2. CMDo shows a list of tasks <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+> Use case ends
+
+### Use case: List all tasks
+
+**MSS**
+
+1. User requests to list all tasks
+2. CMDo shows a list of tasks <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+### Use case: List done tasks
+
+> Use case ends
+
+**MSS**
+
+1. User requests to list done tasks
+2. CMDo shows a list of done tasks <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+> Use case ends
+
+**MSS**
+
+1. User requests to list blocked time slots
+2. CMDo shows a list of blocked time slots <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+> Use case ends
+
+### Use case: Mark a task as done
+
+**MSS**
+
+1. User requests to find a task or list all tasks
+2. CMDo shows a list of tasks 
+3. Mark the task done by index 
+4. The task is marked as done and moved to storage<br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
+
+> 3a1. CMDo shows help message <br>
   Use case resumes at step 2
 
-{More to be added}
+4a. The task will not show up on list or find.
+
+> Use case ends
+
+### Use case: Undo
+
+**MSS**
+
+1. User requests to undo previous action
+2. CMDo requests the user to confirm the action
+3. CMDo undos the previous action <br>
+>Use case ends.
+
+**Extensions**
+
+1a. There are no more previous actions
+
+> Use case ends
+
+2a. User does not confirm the action
+
+> Use case ends
+
+### Use case: Redo
+
+**MSS**
+
+1. User requests to redo previous action
+2. CMDo requests the user to confirm the action
+3. CMDo undos the previous action <br>
+> Use case ends.
+
+**Extensions**
+
+1a. There are no more previous actions
+
+> Use case ends
+
+2a. User does not confirm the action
+
+> Use case ends
+
+### Use case: Change storage location
+
+**MSS**
+
+1. User requests to to change file storage location
+2. CMDo requests the user to confirm the action
+3. CMDo changes the file storage location <br>
+> Use case ends.
+
+**Extensions**
+
+2a. User does not confirm the action
+
+> Use case ends
+
+### Use case: Exit
+
+**MSS**
+
+1. User requests to Exit CMDo
+2. CMDo request the user to confirm the action
+3. CMDo is exited <br>
+>Use case ends.
+
+**Extensions**
+
+2a. User does not confirm the action
+
+> Use case ends
+
 
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java `1.8.0_60` or higher installed.
-2. Should be able to hold up to 1000 persons.
+2. Should be able to hold up to 1000 tasks.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Customize commands to suit user preference
+6. Able to toggle layout for dark and light for colour blindness
+7. Issue reminders for upcoming tasks
+8. Auto-complete functionality for user entries
+9. Google search function in CMDo
```

Comment:
why? do justify ur stand to me and why u need this as a non-func ~ i don't exactly this having google search is a non-func in ur program ~